diff --git a/ziskos/entrypoint/src/alloc/alloc.rs b/ziskos/entrypoint/src/alloc/alloc.rs
index 4a0878e..2d2073a 100644
--- a/ziskos/entrypoint/src/alloc/alloc.rs
+++ b/ziskos/entrypoint/src/alloc/alloc.rs
@@ -133,7 +133,8 @@ pub unsafe extern "C" fn sys_alloc_aligned(bytes: usize, align: usize) -> *mut u
 }
 
 #[cfg(zisk_guest)]
-#[inline(always)]
+#[no_mangle]
+#[inline(never)]
 pub unsafe fn inline_bump_alloc_aligned(bytes: usize, align: usize) -> *mut u8 {
     // SAFETY: Single threaded, so nothing else can touch this while we're working.
     let mut heap_pos = unsafe { HEAP_POS };
diff --git a/ziskos/entrypoint/src/lib.rs b/ziskos/entrypoint/src/lib.rs
index bc8ad6e..94c9c9f 100644
--- a/ziskos/entrypoint/src/lib.rs
+++ b/ziskos/entrypoint/src/lib.rs
@@ -359,6 +359,7 @@ pub mod ziskos {
     use crate::ziskos_definitions::ziskos_config::*;
     use core::arch::asm;
 
+    #[cfg(not(feature = "no_entrypoint"))]
     #[no_mangle]
     #[link_section = ".text.init"]
     unsafe extern "C" fn _start() -> ! {
@@ -417,6 +418,7 @@ pub mod ziskos {
         getrandom::register_custom_getrandom!(zkvm_getrandom);
     }
 
+    #[cfg(not(feature = "no_entrypoint"))]
     #[no_mangle]
     unsafe extern "C" fn _zisk_main() -> i32 {
         {
@@ -613,6 +615,12 @@ pub mod ziskos {
         unimplemented!("sys_argv");
     }
 
+    #[no_mangle]
+    extern "C" fn sys_panic(_msg_ptr: *const u8, _msg_len: usize) -> ! {
+        // Empty implementation - panic handling is done by bflat runtime
+        loop {}
+    }
+
     pub extern "C" fn sys_write_hex(val: usize, ln: bool) {
         let mut buf = [0u8; 19]; // "0x" + 16 hex + \n — stack, no heap
         buf[0] = b'0';
